modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AspNetCoreServerCodegen.java

```diff
@@ -155,6 +219,55 @@ public void processOpts() {
             setAspnetCoreVersion((String) additionalProperties.get(ASPNET_CORE_VERSION));
         }
 
+        if (additionalProperties.containsKey(GENERATE_WWWROOT)) {
+            generateWWWRoot = convertPropertyToBooleanAndWriteBack(GENERATE_WWWROOT);
+        } else {
+            additionalProperties.put(GENERATE_WWWROOT, generateWWWRoot);
+        }
+
+        // CHeck for class modifier if not present set the default value.
+        if (additionalProperties.containsKey(CLASS_MODIFIER)) {
+             classModifier = additionalProperties.get(CLASS_MODIFIER).toString();
+        } else {
+            additionalProperties.put(CLASS_MODIFIER, classModifier);
+        }
+
+        // If class modifierier is abstract then the methods need to be abstrat too.
+        if ("abstract".equals(classModifier)) {
+            operationModifier = classModifier;
+            additionalProperties.put(OPERATION_MODIFIER, operationModifier);
+        }
+
+        if (additionalProperties.containsKey(OPERATION_MODIFIER)) {
+            operationModifier = additionalProperties.get(OPERATION_MODIFIER).toString();
+        } else {
+            additionalProperties.put(OPERATION_MODIFIER, operationModifier);
+        }
+
+        // If operation modifier is abstract then dont generate any body
+        if ("abstract".equals(operationModifier)) {
+            generateBody = false;
+            additionalProperties.put(GENERATE_BODY, generateBody);
+        } else {
+            if (additionalProperties.containsKey(GENERATE_BODY)) {
```

Comment:
should this be **if**/**else if**/**else**

The logic tree seems very weird here

Reply:
The nested if/else is a trivial optimization. I'll move it out to make the logic easier to follow.